Whiskey/Tasks/New-WhiskeyZipArchive.ps1

```diff
@@ -0,0 +1,201 @@
+
+function New-WhiskeyZipArchive
+{
+    [Whiskey.Task("Zip")]
+    [Whiskey.RequiresTool('PowerShellModule::Zip','ZipPath',Version='0.2.*',VersionParameterName='ZipVersion')]
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [Whiskey.Context]
+        $TaskContext,
+
+        [Parameter(Mandatory)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    Import-WhiskeyPowerShellModule -Name 'Zip'
+
+    $archivePath = $TaskParameter['ArchivePath']
+    if( [IO.Path]::IsPathRooted($archivePath) )
+    {
+        $buildRootRegex = '^{0}(\\|/)' -f [regex]::Escape($TaskContext.BuildRoot)
+        if( $archivePath -notmatch $buildRootRegex )
+        {
+            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('ArchivePath: path to ZIP archive "{0}" is outside the build root directory. Please change this path so it is under the "{1}" directory. We recommend using a relative path, as that will always be resolved relative to the build root directory.' -f $archivePath,$TaskContext.BuildRoot)
+            return
+        }
+    }
+    else
+    {
+        $archivePath = Join-Path -Path $TaskContext.BuildRoot -ChildPath $archivePath

```

Comment:
Do we not care to validate that the relative path is inside the build root? E.g. Someone specifies `..\..\File.zip`

Reply:
Added test case.